python/python/lancedb/query.py

```diff
@@ -518,7 +520,34 @@ def to_arrow(self) -> pa.Table:
             vector_column=self._vector_column,
             with_row_id=self._with_row_id,
         )
-        return self._table._execute_query(query)
+
+    def to_arrow(self) -> pa.Table:
+        """
+        Execute the query and return the results as an
+        [Apache Arrow Table](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table).
+
+        In addition to the selected columns, LanceDB also returns a vector
+        and also the "_distance" column which is the distance between the query
+        vector and the returned vectors.
+        """
+        return pa.Table.from_batches(
+            self.to_batches(), self._table._get_query_projected_schema(self.query())
+        )
+
+    def to_batches(self, batch_size: Optional[int] = None) -> Iterable[pa.RecordBatch]:
```

Comment:
Matter of taste, maybe, but I'd prefer `batch_size` is keyword only.

```suggestion
    def to_batches(self, /, batch_size: Optional[int] = None) -> Iterable[pa.RecordBatch]:
```